plugins/formal-verification/formver.viper/src/org/jetbrains/kotlin/formver/viper/Verifier.kt

```diff
@@ -45,7 +45,7 @@ class Verifier {
 
     /** Verify the program.  Returns true on successful verification.
      */
-    fun verify(program: Program, onFailure: (VerifierError) -> Unit): Boolean {
+    fun verify(program: Program, actuallyVerify: Boolean, onFailure: (VerifierError) -> Unit): Boolean {
```

Comment:
I find the `actuallyVerify` argument a bit weird since then you can call the `verify` function without even verifying the program. How about splitting this function into two, one that build the Viper program (and checks for inconsistencies) and another that verifies it.

Reply:
Yeah, that's fair, it is the neater thing.  I've rewritten it that way; we generate the Viper program twice now, but it's not a big deal (at least in the tests the performance effect isn't noticeable).